File: tableify.py
#!/usr/bin/env python

names = 'Adam Brett Cari Donna Ethan Fred Grok Han Izzy Jay'.split()
columns = 3

def tableify(names, columns):
  rows = int(len(names) / columns) + (len(names) % columns > 0)
  result = [[] for _ in range(rows)]

  for i in range(len(names)):
    result[i % rows].append(names[i])
  return result

print str(tableify(names, columns)).replace(', [',",\n [")
 
"""

[["Adam", "Ethan", "Izzy"],
 ["Brett", "Fred", "Jay"],
 ["Cari", "Grok"],
 ["Donna", "Han"]]

"""

File: tableify.ts
#!/usr/bin/env typescript

var list = 'Adam Brett Cari Donna Ethan Fred Grok Han Izzy Jay'.split(' ');
var columns = 3;

function tableify (list: Array<string>, columns: number) {
  var rows = Math.ceil(list.length / columns);
  var result = Array.apply(null, Array(rows)).map(function(){return []})

  for (var i = 0; i < list.length; i++) {
    result[i % rows].push(list[i]);
  }
  return result;
}

console.log(tableify(list, columns));
 
// [ [ 'Adam', 'Ethan', 'Izzy' ],
//   [ 'Brett', 'Fred', 'Jay' ],
//   [ 'Cari', 'Grok' ],
//   [ 'Donna', 'Han' ] ]
